src/main/java/com/exasol/sql/dml/merge/Merge.java

```diff
@@ -71,42 +89,60 @@ public Merge on(final BooleanExpression condition) {
         return this;
     }
 
+    /**
+     * Get the merge condition.
+     *
+     * @return criteria that must be met for the rows in source and destination to be considered a match.
+     */
+    public BooleanExpression getCondition() {
+        return this.condition;
+    }
+
+    /**
+     * Check if the {@code ON} clause exists.
+     *
+     * @return {@code true} if the {@code ON} clause exists.
+     */
+    protected boolean hasCondition() {
+        return this.on != null;
```

Comment:
Similar to above comment, check if `this.on.condition` is not null?

Reply:
After reading the javadoc, maybe you wan to check if `on` clause exists, then rename method to hasOn.

Reply:
Fixed.